identity/src/ed25519.rs

```diff
@@ -170,27 +155,23 @@ impl PublicKey {
 
     /// Try to parse a public key from a byte array containing the actual key as produced by `to_bytes`.
     pub fn try_from_bytes(k: &[u8]) -> Result<PublicKey, DecodingError> {
-        ed25519::PublicKey::from_bytes(k)
+        let k = k[0..32]
```

Comment:
I think this can panic if the slice is < 32 bytes long. We will need to find a different way of coding this. I think arrays have `TryFrom` implementations for slices right? Can we not do something like: `<[u8; 32]>::try_from`?

Reply:
nice catch Thomas! Yup works, thanks, updated!